### PR TITLE
support for automatic generation of `rowKey` or setting from columns

### PR DESCRIPTION
**Pull request by xzdry**
Support for specifying a column as a row key and automatic generation of a row key by default
image

### CODE REVIEW

File: packages/arco-lib/src/widgets/TablePrimaryKeyWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export const TablePrimaryKeyWidget: React.FC<
+  WidgetProps<TablePrimaryKeyWidget, string>
+> = props => {
+  const { value, onChange, component } = props;
+  const { properties } = component;
+  const columns = properties.columns as Static<typeof ColumnSpec>[];
+
+  const keys = ['auto', ...columns.map(c => c.dataIndex)];
```
Comment (tanbowensg) [inline packages/arco-lib/src/widgets/TablePrimaryKeyWidget.tsx:23]: Dose `auto` means a key called `auto` or auto choosing a key? I could be conflict.

File: packages/arco-lib/src/components/Table/Table.tsx [hunk 2]
```diff
@@ -196,6 +206,7 @@ export const Table = implementRuntimeComponent({
 
   useEffect(() => {
     mergeState({ pageSize });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (tanbowensg) [inline packages/arco-lib/src/components/Table/Table.tsx:209]: Can we make sure the `pageSize` will never change or don't need to mergeState when it changes?

File: packages/arco-lib/src/components/Table/Table.tsx [hunk 3]
```diff
@@ -231,13 +242,39 @@ export const Table = implementRuntimeComponent({
...
       // If the `useCustomPagination` is true, then no pagination will be done and data over the pagesize will be sliced
       // Otherwise it will automatically paginate on the front end based on the current page
-      return sortedData?.slice(0, pageSize);
+      data = sortedData?.slice(0, pageSize);
     }
-    return sortedData;
-  }, [pageSize, sortedData, enablePagination, useCustomPagination]);
+
+    // auto-generated row key
+    return rowKey === 'auto'
```
Comment (tanbowensg) [inline packages/arco-lib/src/components/Table/Table.tsx:253]: Do we need to auto generate key for users? I think it rarely happens.

**Review by tanbowensg — CHANGES_REQUESTED** [review]

File: packages/arco-lib/src/widgets/TablePrimaryKeyWidget.tsx (hunk 1 above)
Comment (xzdry) [reply packages/arco-lib/src/widgets/TablePrimaryKeyWidget.tsx:23]: auto means add a key to each item of data

File: packages/arco-lib/src/components/Table/Table.tsx (hunk 3 above)
Comment (xzdry) [reply packages/arco-lib/src/components/Table/Table.tsx:253]: If a default column is not unique as a rowKey, this can also lead to duplicate selections.

Comment (tanbowensg) [reply packages/arco-lib/src/components/Table/Table.tsx:253]: Yes, but I think it should leave to users to decide which is the `rowKey` and make sure it is unique.
Users don't know the rule of auto generating `rowKey`, so even we generated it for user, they won't know what dost it look like.

File: packages/arco-lib/src/components/Table/Table.tsx (hunk 2 above)
Comment (xzdry) [reply packages/arco-lib/src/components/Table/Table.tsx:209]: This is only used to synchronise the initial state of pageSize

File: packages/arco-lib/src/components/Table/Table.tsx (hunk 3 above)
Comment (xzdry) [reply packages/arco-lib/src/components/Table/Table.tsx]: Removed

**Review by tanbowensg — APPROVED** [review]
LGTM

**Pull request merged**